bentoml/_internal/models/model.py

```diff
@@ -352,6 +392,36 @@ def from_dict(data: dict[str, str | dict[str, str]] | ModelContext) -> ModelCont
 
 @attr.frozen
 class ModelSignature:
+    """
+    A model signature represents a method on a model object that can be called.
+
+    Note that anywhere a ``ModelSignature`` is used, a ``dict`` with keys corresponding to the fields
+    can be used instead. For example, instead of ``{"predict": ModelSignature(batchable=True)}``, one
+    can pass ``{"predict": {"batchable": True}}``.
+
+    Fields:
+        batchable:
+            Whether the method should support batching. Note that batching must also
+            be explicitly enabled at runtime in the ``Model.to_runner`` or ``bentoml.runner`` call.
+        batch_dim:
+            The dimension(s) that contain multiple data. For example, if when passing two input arrays
```

Comment:
Maybe add a bit language that's familiar to data scientist here? something like:

`batch_dim` represents the batch dimension in a multidimensional array(numpy.ndarray or tf.Tensor). BentoML uses this information to do dynamic batching of input data.

For example, with `batch_dim=0`, two separate input array `[[1,2]]` and `[[3,4]]` will be grouped along axis 0, into a larger batch input `[[1,2], [3,4]]`.  And similarly with `batch_dim=1`, the two separate input array `[[1], [2]]` and `[[3], [4]]` will be grouped as `[[1,3], [2,4]]`.